src/expr.rs

```diff
@@ -781,10 +791,26 @@ fn rewrite_if_else(context: &RewriteContext,
                                else_block.span.lo);
         let after_else_comment = extract_comment(after_else, &context, offset, width);
 
+        let between_sep = match context.config.else_if_brace_style {
+            ElseIfBraceStyle::AlwaysNextLine |
+            ElseIfBraceStyle::ClosingNextLine => alt_separator.as_str(),
+            _ => " ",
+        };
+        let after_sep = match context.config.else_if_brace_style {
+            ElseIfBraceStyle::AlwaysNextLine => {
+                if last_in_chain {
```

Comment:
Nit: you could use this as a guard for the `ElseIfBraceStyle::AlwaysNextLine` pattern.